File: example/src/main/java/com/thesurix/example/gesturerecycler/model/Month.java
package com.thesurix.example.gesturerecycler.model;


import android.support.annotation.DrawableRes;

public class Month implements MonthItem {

    private String mName;
    private int mDrawableId;

    public Month(final String name, @DrawableRes final int drawableId) {
        mName = name;
        mDrawableId = drawableId;
    }

    @Override
    public MonthItemType getType() {
        return MonthItemType.MONTH;
    }

    @Override
    public String getName() {
        return mName;
    }

    public int getDrawableId() {
        return mDrawableId;
    }
}

File: example/src/main/java/com/thesurix/example/gesturerecycler/model/Month.kt
package com.thesurix.example.gesturerecycler.model


import androidx.annotation.DrawableRes

class Month(override val name: String, @param:DrawableRes val drawableId: Int) : MonthItem {

    override val type: MonthItem.MonthItemType
        get() = MonthItem.MonthItemType.MONTH
}
